### ISSUE TITLE
Using custom maxLevel and disabling marks in Heading causes paste error for certain content

### ISSUE THREAD

**Issue by kiejo** [ProseMirror/prosemirror#340]
I'm using a custom schema with a heading like this:
```javascript
var schemaSpec = defaultSchema.spec.update({
  heading: class extends Heading {
    get maxLevel() { return 3 }
    get containsMarks() { return false }
  }
});

var schema = new Schema(schemaSpec);
```

Copying and pasting all content from this [simple Google Doc](https://docs.google.com/document/d/1ddW_VRaOyzsM5MZUeVgq416byoq9I614Oqdvxy9nuuw) causes the following error:  
![paste_error](https://cloud.githubusercontent.com/assets/152507/15174690/34fb3d1c-1763-11e6-9497-d2eb4457231c.png)

Interestingly, the error only occurs when copying all content. Separately copying and pasting "Heading 2" or "Normal" does not cause any errors (but results in an incorrect formatting, which is probably another issue that might be limited to Google Docs).

**Comment by marijnh** [comment]
(The custom maxLevel appears to have nothing to do with this.)

I've fixed this in the branch for #220, which I'm about to land. The situation in the master branch is, when it comes to ensuring valid node content, a bit of a mess. Is it okay for you to have to port to the new API (which I'll describe when I land the branch) to have access to the fix?

**Comment by kiejo** [comment]
Thanks for the fast response. Having to port to the new API is okay for me. Can't wait for the branch to land!

**Comment by marijnh** [comment]
I've merged the work for #220, this seems to be fine now.
